### ISSUE TITLE
Allow passing client as an option

### ISSUE THREAD

**Issue by jackh726** [nearform/graphql-hooks#374]
Similar issue here: https://github.com/nearform/graphql-hooks/issues/315

### Description
As in the linked issue, I need to make requests to different endpoints. The solution in the linked issue was to just add a new HOC that wraps the current component in a `ClientContext.Provider`. While this does work, it more or less goes contrary to the goals of hooks (at this point, you now have a HOC AND hooks, so why not just go with a HOC only?). Also, if you have to make multiple requests in the same component to different endpoints, it's not even possible in this manner.

### Suggested implementation
The easiest (and imo more elegant) solution would be to just allow passing a client as an option.
For example, in useQuery:
```diff
- const client = React.useContext(ClientContext)
+ const client = opts.client || React.useContext(ClientContext)
```
Then, the user can decide how they get their client (e.g. a separate Context).

All in all, it's a very simple change, but allows multiple endpoints rather easily.

**Comment by bmullan91** [comment]
@jackh726 Sorry for the lack of response on this - this sounds like a good first issue, is this something you'd be interested in taking on?

**Comment by helloguille** [comment]
I'm also interested in this. I need it urgently. I already worked on some small contributions to the library (a head start for GET support, that was actually re-written by @bmullan91 :D).

**Comment by bmullan91** [comment]
@helloguille Feel free to open a PR and I'll take a look 👀

**Comment by bmullan91** [comment]
Available in `graphql-hooks@4.4.3`
